lib/k8s/client/mint/http_adapter.ex

```diff
@@ -455,4 +450,23 @@ defmodule K8s.Client.Mint.HTTPAdapter do
 
     {:noreply, state}
   end
+
+  @spec shutdown_if_closed(t()) :: {:noreply, t()} | {:stop, {:shutdown, :closed}, t()}
+  defp shutdown_if_closed(state) do
+    if Mint.HTTP.open?(state.conn) or any_non_empty_buffers?(state) do
```

Comment:
It seems like this condition is repeated above. Would it be worth putting this into a function? Then you can also explain why the server is not shutdown even if mint has closed the connection, but there are buffers